text/0000-bee-model-crate/0000-bee-model-crate.md

```diff
@@ -299,8 +296,6 @@ Without any bee-model crate, nodes can not exchange transactions. Therefore this
 
 - Where should the actual Transaction struct be initialized/returned? Current ideas are to put it in the build() of the TransactionBuilder or as it currently is in the compute() of Pow.
 
-- Where should Transaction fields get validated? One-time validation once build() is called by the separate *TransactionBuilderValidator* as it currently is, or use **setters** in the TransactionBuilder and validate in each setter separately?
-
-- Should we use setters for the TransactionBuilder?
+- Should we introduce a fluent API to the TransactionBuilder?
 
-- Introducing a fluent API to the TransactionBuilder?
+- Should we use Option<T> for the fields in TransactionBuilder? 
```

Comment:
Some fields would need this I guess. Others like timestamp, index etc are mandatory.

Reply:
The `build` method is responsible for ensuring that mandatory fields are set, not the type of the field.